codegen/src/main/java/software/amazon/awssdk/codegen/poet/rules/SymbolTable.java

```diff
@@ -13,22 +13,23 @@
  * permissions and limitations under the License.
  */
 
-package software.amazon.awssdk.codegen.poet.rules2;
+package software.amazon.awssdk.codegen.poet.rules;
 
 import java.util.Collections;
 import java.util.LinkedHashMap;
 import java.util.Map;
+import software.amazon.awssdk.codegen.model.rules.endpoints.BuiltInParameter;
 import software.amazon.awssdk.utils.Validate;
 
 public final class SymbolTable {
     private final Map<String, RuleType> params;
     private final Map<String, RuleType> locals;
-    private final String regionParamName;
+    private final Map<String, BuiltInParameter> builtInParamTypes;
 
     SymbolTable(Builder builder) {
         this.params = Collections.unmodifiableMap(new LinkedHashMap<>(builder.params));
         this.locals = Collections.unmodifiableMap(new LinkedHashMap<>(builder.locals));
-        this.regionParamName = builder.regionParamName;
+        this.builtInParamTypes = Collections.unmodifiableMap(new LinkedHashMap<>(builder.builtInParamTypes));
```

Comment:
Do we need the overhead of maintaining a list with the LinkedHashMap ? I do not see us needing element ordering here in this class. Could we just do HashMap instead ?

Reply:
Seems reasonable. Let me see if there's any tests that assume order.